examples/matching-engine/tests/basic_operations.rs

```diff
@@ -0,0 +1,371 @@
+
+    // Create fungible tokens
+    let fungible_module_id_a = user_chain_a
+        .publish_bytecode_files_in::<fungible::FungibleTokenAbi, fungible::Parameters, fungible::InitialState>("../fungible")
+        .await;
+    let fungible_module_id_b = user_chain_b
+        .publish_bytecode_files_in::<fungible::FungibleTokenAbi, fungible::Parameters, fungible::InitialState>("../fungible")
+        .await;
+
+    // Give users enough tokens to cover test scenarios
+    // For bids with price*quantity, users need sufficient token0
+    let initial_state_a =
+        fungible::InitialStateBuilder::default().with_account(owner_a, Amount::from_tokens(10000));
+    let params_a = fungible::Parameters::new("TokenA");
+    let token_id_a = user_chain_a
+        .create_application(
+            fungible_module_id_a,
+            params_a,
+            initial_state_a.build(),
+            vec![],
+        )
+        .await;
+
+    let initial_state_b =
+        fungible::InitialStateBuilder::default().with_account(owner_b, Amount::from_tokens(10000));
+    let params_b = fungible::Parameters::new("TokenB");
+    let token_id_b = user_chain_b
+        .create_application(
+            fungible_module_id_b,
+            params_b,
+            initial_state_b.build(),
+            vec![],
+        )
+        .await;
+
+    // Create matching engine
+    let matching_parameter = Parameters {
+        tokens: [token_id_a, token_id_b],
+    };
+    let matching_id = matching_chain
+        .create_application(
+            module_id,
+            matching_parameter,
+            (),
+            vec![token_id_a.forget_abi(), token_id_b.forget_abi()],
+        )
+        .await;
+
+    (
+        validator,
+        matching_id,
+        token_id_a,
+        token_id_b,
+        user_chain_a,
+        user_chain_b,
+        matching_chain,
+    )
+}
+
+#[tokio::test]
+async fn test_insert_bid_order() {
+    let (
+        _validator,
+        matching_id,
+        _token_id_a,
+        _token_id_b,
+        user_chain_a,
+        _user_chain_b,
+        matching_chain,
+    ) = setup_matching_engine().await;
+
+    let owner_a = AccountOwner::from(user_chain_a.public_key());
+
+    // Insert a bid order
+    let order = Order::Insert {
+        owner: owner_a,
+        amount: Amount::from_tokens(10),
+        nature: OrderNature::Bid,
+        price: Price { price: 100 },
+    };
+
+    let operation = Operation::ExecuteOrder { order };
+    let certificate = user_chain_a
+        .add_block(|block| {
+            block.with_operation(matching_id, operation);
+        })
+        .await;
+
+    // Process the order on the matching engine chain
+    matching_chain
+        .add_block(|block| {
+            block.with_messages_from(&certificate);
+        })
+        .await;
+
+    user_chain_a.handle_received_messages().await;
+    matching_chain.handle_received_messages().await;
+
+    // The order should be created successfully (no panics)
+}
```

Comment:
(I'm not a fan of having tests that are basically subsets of another test. But I guess these ones are cheap.)

Reply:
No actually these tests are surprisingly slow (perhaps because of app linking time?)